testing/README.md

```diff
@@ -884,3 +884,78 @@ Deno.test("secondInterval calls callback every second and stops after being clea
   }
 });
 ```
+
+## Conditional Type Checks
+
+As TypeScript's type system becomes more complex it's useful to be able to write
+tests for what a type should be.
+
+The `testing/types.ts` module offers reusable conditional types to do these
+checks.
+
+### Type Checks
+
+These will resolve to the type `true` when they match and `false` otherwise.
+
+- `Has<T, U>` - Checks if `T` has `U`.
+- `NotHas<T, U>` - Checks if `T` does not have `U`.
+- `IsNullable<T>` - Checks if `T` is possibly `null` or `undefined`.
+- `IsExact<T, U>` - Checks if `T` exactly matches `U`.
+- `IsAny<T>` - Checks if `T` is the `any` type.
+- `IsNever<T>` - Checks if `T` is the `never` type.
+- `IsUnknown<T>` - Checks if `T` is the `unknown` type.
+
+### Ways to Test
+
+Use what you prefer:
+
+1. The `AssertTrue`, `AssertFalse`, or `Assert` types.
+2. The `assert` function.
```

Comment:
Do we need to provide these 2 options? I feel the 2. is better for reading and 1. is a little confusing.

cc @dsherret

Reply:
Maybe I'm missing something but I don't see a ton of value in `AssertTrue` and gang. `assertType` seems a lot more intuitive to me.

Reply:
@dsherret sorry to push, but I would hate for this to go stale.

Reply:
Perhaps, it's best to proceed with 2.

Reply:
I'm game to proceed with 2 for now, if it turns out to be confusing we can always deprecate and remove one.